Riot/Modules/Room/DataSources/RoomDataSource.swift

```diff
@@ -198,7 +198,30 @@ extension RoomDataSource {
         return editableTextMessage
     }
 
-    
+    /// Send a voice broadcast to the room.
+    ///
+    /// While sending, a fake event will be echoed in the messages list.
+    /// Once complete, this local echo will be replaced by the event saved by the homeserver.
+    ///
+    /// - Parameters:
+    ///   - audioFileLocalURL: the local filesystem path of the audio file to send.
+    ///   - voiceBroadcastInfoEventId: The id of the voice broadcast info event.
+    ///   - mimeType: (optional) the mime type of the file. Defaults to `audio/ogg`
+    ///   - duration: the length of the voice message in milliseconds
+    ///   - samples: an array of floating point values normalized to [0, 1], boxed within NSNumbers
+    ///   - success: A block object called when the operation succeeds. It returns the event id of the event generated on the homeserver
+    ///   - failure: A block object called when the operation fails.
+    @objc func sendVoiceBroadcast(audioFileLocalURL: URL, voiceBroadcastInfoEventId: String, mimeType: String?, duration: UInt, samples: [Float]?, success:@escaping ((String?) -> Void), failure:@escaping ((Error?) -> Void)) {
```

Comment:
- I would rename this method to clarify its usage
"sendChunkOfVoiceBroadcast" or "broadcastVoiceRecording" (I prefer the first one)
- The description would be "/// Send a bunch of a voice broadcast.
- We should be able to ignore the local echo handling for the chunks of voice, they will be aggregated -> this should help to simplify the code
- I don't think this method is at the right place. I would expect only the methods to manage the voice broadcast itself (the state events `_info`). The send of the chunks should be managed by the current VocieBroadcastService (for that the service should keep a reference to the room in which the broadcast takes place)


BTW, you need to add a mark for this method, it is currently grouped with NSAttributedString sending:
 <img width="407" alt="image" src="https://user-images.githubusercontent.com/8969772/194532667-87c4e156-9ad2-4609-8958-e24ae13c13d1.png">